pkg/cloud-init/cloud-init.go

```diff
@@ -263,18 +329,19 @@ func readCloudInitConfigDriveSource(source *v1.CloudInitConfigDriveSource) (*Clo
 	}, nil
 }
 
-func readCloudInitNoCloudMetaData(name, hostname, namespace string) *Metadata {
-	return &Metadata{
+func readCloudInitNoCloudMetaData(name, hostname, namespace string) *NoCloudMetadata {
+	return &NoCloudMetadata{
 		InstanceID:    fmt.Sprintf("%s.%s", name, namespace),
 		LocalHostname: hostname,
 	}
 }
 
-func readCloudInitConfigDriveMetaData(uid, name, hostname, namespace string) *Metadata {
-	return &Metadata{
-		UUID:       uid,
-		InstanceID: fmt.Sprintf("%s.%s", name, namespace),
-		Hostname:   hostname,
+func readCloudInitConfigDriveMetaData(uid, name, hostname, namespace string, keys map[string]string) *ConfigDriveMetadata {
```

Comment:
I know this is not your change but why this is called `readCloudInitConfigDriveMetaData`?